### PR TITLE
Split du's seive from arithmetic funciton to new chapter, modify typo…

### PR DESCRIPTION
**Pull request by PolarisChiba**
… in linear sieve

### CODE REVIEW

**Comment by harry900831** [comment]
You may split modify typo and add du's seive chapter to different commit or even different PR.

File: src/math/du_seive.md [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+    for (int i = 2; i < lim; ++ i) {
+        if (i % (pf[i] * pf[i]) != 0)
+            mu[i] = -mu[i / pf[i]];
+        psm[i] = psm[i - 1] + mu[i];
+    }
+}
+
+int main() {
+	ll n;
+    cin >> n;
```
Comment (harry900831) [inline src/math/du_seive.md:343]: wrong indention

File: src/math/du_seive.md [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+}
+```
+
+或者也可以使用迭代來實作：
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
```
Comment (harry900831) [inline src/math/du_seive.md:310]: using ll = long long;

File: src/math/du_seive.md [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+        	if (i * p >= lim) break;
+            pf[i * p] = p;
+            if (i % p == 0) break;
+        }
+    }
+
+	vector<int> phi(lim);
+	psp.resize(lim);
+	phi[1] = psp[1] = 1;
+    for (int i = 2; i < lim; ++ i) {
```
Comment (harry900831) [inline src/math/du_seive.md:421]: wrong indention

File: src/math/du_seive.md [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+    else x = s[n];
+    
+    string ans = "";
+    while (x) ans.push_back(x % 10 + '0'), x /= 10;
+    reverse(ans.begin(), ans.end());
+    cout << ans << "\n";
+}
+```
+
+（以上程式碼皆未在任何 OJ 上測試過，因此可能有著潛在 BUG，而且也沒有壓好常數。）
```
Comment (harry900831) [inline src/math/du_seive.md:533]: 可以找一下哪裡可以 submit 測一下你的code嗎？
否則讀者看到這邊 也沒有真的有 OJ 可以練習

File: src/math/du_seive.md [hunk 5]
```diff
@@ -0,0 +1,588 @@
+# 杜教篩
```
Comment (harry900831) [inline src/math/du_seive.md:1]: 有太多 縮排沒縮好的Code

建議使用 clang-format 排好版

**Review by harry900831 — CHANGES_REQUESTED** [review]

File: src/math/du_seive.md (hunk 4 above)
Comment (PolarisChiba) [reply src/math/du_seive.md]: 沒有 OJ，都是假的。

Comment (harry900831) [reply src/math/du_seive.md]: https://www.luogu.com.cn/problem/P4213

不要 resolve 還沒解決的 comment

Comment (harry900831) [reply src/math/du_seive.md]: https://blog.csdn.net/skywalkert/article/details/50500009

**Comment by harry900831** [comment]
先額外發一格 PR 修正 typo  `if (i * p > n) break;`
我先 merge 那部分

File: src/math/du_sieve.md [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+
+Bottleneck 在於求 \\( \mu \\) 的前綴和，時間複雜度依然是 \\( \mathcal{O}(n^\frac{2}{3}) \\)。
+
+（實作略）
+
+## 練習題
+
+>  求\\( \displaystyle S(n)=\sum_{i=1}^n \sum_{j=1}^n ij\times \gcd(i,j) \\) 的值。（\\( n\le 10^{10} \\)）
+
+~~這題一下子就變得好難QQ~~
```
Comment (harry900831) [inline src/math/du_sieve.md:616]: Any references?

File: src/math/du_sieve.md [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+
+因此可以結合例題一還有數論分塊，來求出 \\( \varphi \\) 的前綴和。
+
+Bottleneck 在於求 \\( \mu \\) 的前綴和，時間複雜度依然是 \\( \mathcal{O}(n^\frac{2}{3}) \\)。
+
+（實作略）
+
+## 練習題
+
+>  求\\( \displaystyle S(n)=\sum_{i=1}^n \sum_{j=1}^n ij\times \gcd(i,j) \\) 的值。（\\( n\le 10^{10} \\)）
```
Comment (harry900831) [inline src/math/du_sieve.md:614]: At least give us a hint.

File: src/math/du_sieve.md [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+
+這樣一來，\\( r = \left\lfloor\frac n k \right\rfloor \\)，因此 \\( r\in R(n) \\)。
+
+也就是說，要求的 \\( \sum_{i=1}^r \mu(i) \\)，恰好和杜教篩中會求到的相同。
+
+因此可以結合例題一還有數論分塊，來求出 \\( \varphi \\) 的前綴和。
+
+Bottleneck 在於求 \\( \mu \\) 的前綴和，時間複雜度依然是 \\( \mathcal{O}(n^\frac{2}{3}) \\)。
+
+（實作略）
```
Comment (harry900831) [inline src/math/du_sieve.md:610]: 略過不用特別寫出來

但如果是希望能放的，放個 TODO 在這邊

File: src/math/du_sieve.md [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+
+  string ans = "";
+  while (x)
+    ans.push_back(x % 10 + '0'), x /= 10;
+  reverse(ans.begin(), ans.end());
+  cout << ans << "\n";
+}
+```
+
+（以上程式碼皆未在任何 OJ 上測試過，因此可能有著潛在 BUG，而且也沒有壓好常數。）
```
Comment (harry900831) [inline src/math/du_sieve.md:561]: :)

**Review by harry900831 — CHANGES_REQUESTED** [review]

File: src/math/du_seive.md (hunk 4 above)
Comment (PolarisChiba) [reply src/math/du_seive.md]: 不要，這些地方我不熟，我不寫我不熟或不確定的東西。

File: src/SUMMARY.md [hunk 10]
```diff
@@ -13,6 +13,7 @@
 - [Flow]()
 - [Mathematics](math/intro.md)
 	- [Arithmetic Function](math/arithmetic_function.md)
```
Comment (harry900831) [inline src/SUMMARY.md:15]: 這裡要改成 兩個空白 不然我無法使用 mdbook 顯示

**Review by harry900831 — CHANGES_REQUESTED** [review]

File: src/math/du_seive.md (hunk 4 above)
Comment (harry900831) [reply src/math/du_seive.md]: 我只是提供你題目

File: src/math/du_sieve.md (hunk 6 above)
Comment (PolarisChiba) [reply src/math/du_sieve.md]: 我沒有。

Comment (PolarisChiba) [reply src/math/du_sieve.md]: 我認為就權當式子堆導的練習就好。

File: src/math/arithmetic_decomposition.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
+# 數論分塊（整除分塊）
```
Comment (harry900831) [inline src/math/arithmetic_decomposition.md:1]: arithmetic decomposition looks weird to me.

File: src/math/arithmetic_decomposition.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+觀察到，數值相同的數字會是連續的，我們稱呼極大的連續相同數字為一「塊」。
+
+舉例來說：\\( [1, 1] \\)、\\( [2, 2] \\)、\\( [3, 3] \\)、\\( [4, 5] \\)、\\( [6, 10] \\) 各是一塊。
+
+而我們會有以下定理：
+
+> \\( i \\) 所在的塊（值為 \\( \lfloor n/i\rfloor \\) 的塊）的右端點為 \\( \lfloor n/\lfloor n/i\rfloor\rfloor \\)。
+
+舉例來說：\\( n=10 \\) 的情況下，\\( 7 \\) 所在的塊的右端點是 \\( \lfloor 10/\lfloor10/7\rfloor\rfloor=\lfloor10/1\rfloor=10 \\)。
```
Comment (harry900831) [inline src/math/arithmetic_decomposition.md:59]: use \frac for all \lfloor x/\lfloory/z\rfloor\rfloor

Please fix all the followings.

File: src/math/arithmetic_decomposition.md (hunk 11 above)
Comment (harry900831) [reply src/math/arithmetic_decomposition.md:1]: https://codeforces.com/blog/entry/53925
https://en.wikipedia.org/wiki/Divisor_summatory_function

File: src/math/arithmetic_decomposition.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+  ans += 1LL * (r - i + 1) * (n / i) * (m / i);
+}
+```
+
+## 練習題
+
+* [Atcoder ABC 230 E](https://atcoder.jp/contests/abc230/tasks/abc230_e)
+* [Codeforces 1263 C](https://codeforces.com/problemset/problem/1263/C)
+* [Codefroces 616 E](https://codeforces.com/problemset/problem/616/E)
+* [Codeforces 1780 E](https://codeforces.cc/contest/1780/problem/E)
```
Comment (harry900831) [inline src/math/arithmetic_decomposition.md:214]: Could you provide some hint or tag for these problem?

File: src/math/arithmetic_decomposition.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
+# 數論分塊（整除分塊）
+
+## 例題
```
Comment (harry900831) [inline src/math/arithmetic_decomposition.md:3]: Please revise your title hierarchy. It looks weird to me now.

File: src/math/arithmetic_decomposition.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+```cpp
+for (int i = 1, r = 0; i <= n; i = r + 1) {
+  r = n / (n / i);
+  ans += 1LL * (r - i + 1) * (n / i);
+}
+```
+
+由於不同數值只有 \\( \mathcal{O}(\sqrt{n}) \\) 個，而每個不同數值也只會被算到一次。
+
+因此時間複雜度從 \\(O(n)\\) 降為 \\(O(\sqrt{n})\\)。
```
Comment (harry900831) [inline src/math/arithmetic_decomposition.md:101]: https://math.stackexchange.com/questions/487401/sum-of-floor-of-harmonic-progression-sum-i-1n-lfloor-frac-ni-rfloor-2-sum

Consider adding this solution too

File: src/math/arithmetic_decomposition.md (hunk 11 above)
Comment (harry900831) [reply src/math/arithmetic_decomposition.md:1]: I didn't find any proper English title for this chapter, maybe call it 數論分塊 or 整除分塊 (I prefer 整除分塊)

File: src/math/arithmetic_decomposition.md (hunk 14 above)
Comment (harry900831) [reply src/math/arithmetic_decomposition.md:3]: maybe:

# 整除分塊

## introduction
### problem
### right end point

## application for 整除分塊
### 捲積的前綴和
### 複合數論分塊
## Exercises

**Review by harry900831 — CHANGES_REQUESTED** [review]

**Comment by harry900831** [comment]
I will probably merge this first after this revision.

File: src/math/arithmetic_decomposition.md (hunk 15 above)
Comment (PolarisChiba) [reply src/math/arithmetic_decomposition.md:101]: it seems unpractical to me.

File: src/math/arithmetic_decomposition.md (hunk 13 above)
Comment (PolarisChiba) [reply src/math/arithmetic_decomposition.md:214]: no

Comment (PolarisChiba) [reply src/math/arithmetic_decomposition.md:214]: the first three are too basic.
I can't solve the last one.

**Review by harry900831 — CHANGES_REQUESTED** [review]

**Review by harry900831 — APPROVED** [review]

**Pull request merged**